arch/arm/soc/st_stm32/stm32f4/soc_pinmux.c

```diff
@@ -15,32 +15,63 @@
 #define PAD(AF, func)				\
 				[AF - 1] = func
 
+#define PINMUX_PWM(port, chan, pin)		\
+	PAD(STM32F4_PINMUX_FUNC_##pin##_PWM##port##_CH##chan,\
+	    STM32F4X_PIN_CONFIG_AF_PUSH_UP),
+
+#define PINMUX_UART(port, dir, pin)		\
+	PAD(STM32F4_PINMUX_FUNC_##pin##_##port##_##dir, \
+	    STM32F4X_PIN_CONFIG_AF_PUSH_UP),
+
+/* Blank pinmux by default */
+#define PINMUX_PWM2(pin, chan)
+#define PINMUX_UART1(dir, pin)
+#define PINMUX_UART2(dir, pin)
+
+#ifdef CONFIG_PWM_STM32_2
+	#undef PINMUX_PWM2
+	#define PINMUX_PWM2(pin, chan)		PINMUX_PWM(2, chan, pin)
+#endif
+
+#ifdef CONFIG_UART_STM32_PORT_1
+	#undef PINMUX_UART1
+	#define PINMUX_UART1(dir, pin)		PINMUX_UART(USART1, dir, pin)
+#endif
+
+#ifdef CONFIG_UART_STM32_PORT_2
+	#undef PINMUX_UART2
+	#define PINMUX_UART2(dir, pin)		PINMUX_UART(USART2, dir, pin)
+#endif
+
+#define PINMUX_UART_TX(pin, port)		PINMUX_UART##port(TX, pin)
+#define PINMUX_UART_RX(pin, port)		PINMUX_UART##port(RX, pin)
+
 static const stm32_pin_func_t pin_pa0_funcs[] = {
```

Comment:
This is an interesting proposal (that would eventually deserve to be extended to whole stm32 family to keep consistency across STM32 series). Thought, please keep it easily readable and consistent across IPs. For instance:
PINMUX_UART(PA2, UART2, TX) 
PINMUX_PWM(PA0, PWM2, CH1)
Aim is to understand what macro does at first reading w/o refering to macro definition.